src/inet/TCPEndPoint.cpp

```diff
@@ -2446,15 +2463,23 @@ void TCPEndPoint::ReceiveData()
         // Otherwise, add the new data onto the receive queue.
         else if (isNewBuf)
         {
-            rcvBuf->SetDataLength(rcvBuf->DataLength() + (uint16_t) rcvLen);
+            assert(rcvLen > 0);
```

Comment:
Why a straight-up C-style assert here rather than the VerifyOrDie macro?

Reply:
I didn't know about `VerifyOrDie`.  Filed https://github.com/project-chip/connectedhomeip/issues/2637 and will fix.